### PR TITLE
Problem: (CRO-247) Transaction broadcast is prone to DoS attacks

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Added maximum size of binary encoded data for transaction and also put limits on number of inputs

### CODE REVIEW

File: chain-core/src/state/account.rs [hunk 1]
```diff
@@ -19,6 +19,14 @@ use secp256k1::recovery::{RecoverableSignature, RecoveryId};
 use std::convert::{From, TryFrom};
 use std::fmt;
 
+const MAX_INPUTS_LENGTH: usize = 64;
```
Comment (leejw51crypto) [inline chain-core/src/state/account.rs]: for exchange,
input length can be very large.

Comment (leejw51crypto) [reply chain-core/src/state/account.rs]: it will be filtered out by size check,
so how about using very large count ?

Comment (leejw51crypto) [reply chain-core/src/state/account.rs]: or no limit

File: chain-core/src/state/account.rs [hunk 2]
```diff
@@ -186,14 +194,41 @@ impl StakedStateOpAttributes {
...
     pub attributes: StakedStateOpAttributes,
 }
 
+impl Decode for DepositBondTx {
+    fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
+        let size = input
+            .remaining_len()?
+            .ok_or_else(|| "Unable to calculate size of input")?;
+
+        if size > MAX_DEPOSIT_TX_SIZE {
```
Comment (leejw51crypto) [inline chain-core/src/state/account.rs]: because utxo table is vector, it can be filered prior to deserialization

Comment (leejw51crypto) [reply chain-core/src/state/account.rs]: what do you think ?  @tomtau

File: chain-core/src/state/account.rs (hunk 1 above)
Comment (tomtau) [reply chain-core/src/state/account.rs]: this is a limit for 1 transaction -- if more inputs are needed, one can submit more transactions? @leejw51crypto 
@calvinaco any thoughts on reasonable limits?

File: chain-core/src/state/account.rs (hunk 2 above)
Comment (tomtau) [reply chain-core/src/state/account.rs]: do you mean to find the length as an offset in the binary?

File: chain-core/src/state/account.rs (hunk 1 above)
Comment (leejw51crypto) [reply chain-core/src/state/account.rs]: very large utxo txs will be filtered by the size.
so limiting utxo count doesn't have much effect.

for the certain circumstances, for massive test,
large utxo in tx is necessary.
for stress test etc.

Comment (leejw51crypto) [reply chain-core/src/state/account.rs]: we can use many txs though.

File: chain-core/src/state/account.rs (hunk 2 above)
Comment (leejw51crypto) [reply chain-core/src/state/account.rs]: yes,
if vector is really big, 
total message size is also big,
then it will be already filtered out.

Comment (leejw51crypto) [reply chain-core/src/state/account.rs]: if we apply limit of individual vectors,
then program can be more complicated.

**Pull request merged**